org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/JavaElementPropertyTester.java

```diff
@@ -87,8 +88,13 @@ public class JavaElementPropertyTester extends PropertyTester {
 	public static final String PROJECT_OPTION = "projectOption"; //$NON-NLS-1$
 
 
+	@SuppressWarnings("boxing")
 	@Override
 	public boolean test(Object receiver, String method, Object[] args, Object expectedValue) {
+		return JavaCore.call(() -> testCaches(receiver, method, args, expectedValue));
+	}
+
+	public boolean testCaches(Object receiver, String method, Object[] args, Object expectedValue) {
```

Comment:
Should this be private and named `testCached` ?

Reply:
exactly. thanks